build-logic/basics/src/main/kotlin/gradlebuild/basics/BuildParams.kt

```diff
@@ -279,8 +284,8 @@ val Project.rerunAllTests: Provider<String>
     get() = gradleProperty(RERUN_ALL_TESTS)
 
 
-val Project.testJavaVendor: Provider<String>
-    get() = propertyFromAnySource(TEST_JAVA_VENDOR)
+val Project.testJavaVendor: Provider<JvmVendorSpec>
+    get() = propertyFromAnySource(TEST_JAVA_VENDOR).map { VENDOR_MAPPING.getValue(it) }
```

Comment:
```suggestion
val Project.testJavaVendor: Provider<JvmVendorSpec> = propertyFromAnySource(TEST_JAVA_VENDOR).map { VENDOR_MAPPING.getValue(it) }
```

Does this need to be lazy? `Provider` is already lazy, this should work as well, right?